src/payjoin_handler.rs

```diff
@@ -0,0 +1,85 @@
+use crate::types::{ChannelManager, Wallet};
+use bdk::SignOptions;
+use bitcoin::psbt::Psbt;
+use bitcoin::secp256k1::PublicKey;
+use bitcoin::ScriptBuf;
+use lightning::ln::ChannelId;
+use lightning::util::persist::KVStore;
+use lightning_payjoin::error::Error as PayjoinError;
+use lightning_payjoin::scheduler::ChannelScheduler;
+use lightning_payjoin::{LightningPayjoin, Receiver};
+use std::sync::Arc;
+use tokio::sync::Mutex;
+
+pub(crate) struct PayjoinHandler<K: KVStore + Sync + Send + 'static> {
+	inner: Arc<Mutex<LightningPayjoin<PayjoinReceiver<K>>>>,
+}
+
+impl<K: KVStore + Sync + Send + 'static> PayjoinHandler<K> {
+	pub(crate) fn new(
+		wallet: Arc<Wallet>, channel_manager: Arc<ChannelManager<K>>,
+		scheduler: Arc<Mutex<ChannelScheduler>>,
+	) -> Self {
+		Self {
+			inner: Arc::new(Mutex::new(LightningPayjoin::new(
+				PayjoinReceiver::new(wallet, channel_manager),
+				scheduler,
+			))),
+		}
+	}
+
+	pub(crate) async fn serve(&self, stream: tokio::net::TcpStream) -> Result<(), PayjoinError> {
+		self.inner.lock().await.serve_incoming_http_request(stream).await
+	}
+}
+
+struct PayjoinReceiver<K: KVStore + Sync + Send + 'static> {
+	channel_manager: Arc<ChannelManager<K>>,
+	wallet: Arc<Wallet>,
+}
+
+impl<K> Clone for PayjoinReceiver<K>
+where
+	K: KVStore + Sync + Send + 'static,
+{
+	fn clone(&self) -> Self {
+		Self { channel_manager: self.channel_manager.clone(), wallet: self.wallet.clone() }
+	}
+}
+
+impl<K: KVStore + Sync + Send + 'static> PayjoinReceiver<K> {
+	fn new(wallet: Arc<Wallet>, channel_manager: Arc<ChannelManager<K>>) -> Self {
+		Self { wallet, channel_manager }
+	}
+}
+
+impl<K: KVStore + Sync + Send + 'static> Receiver for PayjoinReceiver<K> {
+	fn check_inputs_not_owned(&self, script: &ScriptBuf) -> Result<bool, PayjoinError> {
+		Ok(self.wallet.is_mine(script)?)
+	}
+
+	fn identify_my_outputs(&self, script: &ScriptBuf) -> Result<bool, PayjoinError> {
+		Ok(self.wallet.is_mine(script)?)
+	}
+
+	fn funding_transaction_generated(
+		&self, temporary_channel_id: &ChannelId, counterparty_node_id: PublicKey,
+		funding_tx: bitcoin::Transaction,
+	) -> Result<(), PayjoinError> {
+		Ok(self.channel_manager.funding_transaction_generated(
+			temporary_channel_id,
+			&counterparty_node_id,
+			funding_tx,
+		)?)
+	}
+
+	fn sign_tx(&self, psbt: &Psbt) -> Result<Psbt, PayjoinError> {
+		let mut sign_options = SignOptions::default();
+		sign_options.try_finalize = false;
+		Ok(self.wallet.sign_tx(psbt, Some(sign_options))?)
+	}
+
+	fn can_broadcast(&self, _transaction: &bitcoin::Transaction) -> Result<bool, PayjoinError> {
+		Ok(true)
```

Comment:
oops, this should be fixed. I dont think the current blockchain or wallet implementation in ldk-node have this functionality. basically need equivalent to https://developer.bitcoin.org/reference/rpc/testmempoolaccept.html.
I think its achievable with `consensus::verify` module in `bitcoin`